File: hello-world/hello_jl.rb
#!/usr/bin/env ruby
#= Code here is only visible to Ruby

puts "Hello from \x1b[91mRuby\x1b[m!"

=begin # Code here is only visible to Julia =#
puts = println
puts("Hello from \x1b[35mJulia\x1b[m!")

#=
=end # Code here is visible to Julia and Ruby =#
puts("Hello from \x1b[35mJulia\x1b[m and \x1b[91mRuby\x1b[m!")

File: hello-world/hello_jl.lua
#!/usr/bin/env lua
_=#_G--[[
0 # Code here is only visible to Julia

println("Hello from \x1b[35mJulia\x1b[m!")

#= Code here is only visible to Lua ]]

local println = print
println("Hello from \x1b[94mLua\x1b[m!")

-- Code here is visible to Lua and Julia =#

println("Hello from \x1b[94mLua\x1b[m and \x1b[35mJulia\x1b[m!")
